src/view/configuration/edgeConfigurationsSection.jsx

```diff
@@ -0,0 +1,397 @@
+          instanceValues.edgeConfigFreeformInputMethod.developmentEdgeConfigId;
+      }
+    }
+    return instanceSettings;
+  },
+  instanceValidationSchema: object()
+    .shape({
+      edgeConfigSelectInputMethod: object().when("edgeConfigInputMethod", {
+        is: INPUT_METHOD.SELECT,
+        then: object().shape({
+          edgeConfig: object().required("Please specify a datastream.")
+        })
+      }),
+      edgeConfigFreeformInputMethod: object().when("edgeConfigInputMethod", {
+        is: INPUT_METHOD.FREEFORM,
+        then: object().shape({
+          edgeConfigId: string().required("Please specify a datastream.")
+        })
+      })
+    })
+    // TestCafe doesn't allow this to be an arrow function because of
+    // how it scopes "this".
+    // eslint-disable-next-line func-names
+    .test("uniqueEdgeConfigId", function(instance, testContext) {
+      const { path: instancePath, parent: instances } = testContext;
+
+      const getEdgeConfigIdFromInstance = inst => {
+        return inst.edgeConfigInputMethod === INPUT_METHOD.SELECT
+          ? inst.edgeConfigSelectInputMethod.edgeConfig?.id
+          : inst.edgeConfigFreeformInputMethod.edgeConfigId;
+      };
+
+      const edgeConfigId = getEdgeConfigIdFromInstance(instance);
+
+      if (!edgeConfigId) {
+        return true;
+      }
+
+      const firstInstanceWithSameEdgeConfigId = instances.find(
+        candidateInstance =>
+          getEdgeConfigIdFromInstance(candidateInstance) === edgeConfigId
+      );
+      const isDuplicate = firstInstanceWithSameEdgeConfigId !== instance;
+
+      const edgeConfigIdFieldName =
+        instance.edgeConfigInputMethod === INPUT_METHOD.SELECT
+          ? `edgeConfigSelectInputMethod.edgeConfig.id`
+          : `edgeConfigFreeformInputMethod.edgeConfigId`;
+
+      return (
+        !isDuplicate ||
+        this.createError({
+          path: `${instancePath}.${edgeConfigIdFieldName}`,
+          message:
+            "Please provide a value unique from those used for other instances."
+        })
+      );
+    })
+};
+
+const EdgeConfigurationsSection = ({ instanceFieldName }) => {
+  const [{ value: inputMethod }] = useField(
+    `${instanceFieldName}.edgeConfigInputMethod`
+  );
+
+  return (
+    <>
+      <SectionHeader learnMoreUrl="https://adobe.ly/3eY91Er">
+        Datastreams
+      </SectionHeader>
+      <FormElementContainer>
+        <RadioGroup
+          label="Input Method"
+          name={`${instanceFieldName}.edgeConfigInputMethod`}
+          orientation="horizontal"
+        >
+          <Radio
+            data-test-id="edgeConfigInputMethodSelectRadio"
+            value={INPUT_METHOD.SELECT}
+          >
+            Choose from list
+          </Radio>
+          <Radio
+            data-test-id="edgeConfigInputMethodFreeformRadio"
+            value={INPUT_METHOD.FREEFORM}
+          >
+            Enter values
+          </Radio>
+        </RadioGroup>
```

Comment:
Only the first instance should have this radio group. We cannot fetch the data streams for the second instance because the IMS org will be different.